src/goto-instrument/cover_instrument_branch.cpp

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************\
+
+Module: Coverage Instrumentation
+
+Author: Daniel Kroening
+
+\*******************************************************************/
+
+/// \file
+/// Coverage Instrumentation for Branches
+
+#include "cover_instrument.h"
+#include "cover_basic_blocks.h"
+
+void cover_branch_instrumentert::instrument(
+  goto_programt &goto_program,
+  goto_programt::targett &i_it,
+  const cover_basic_blockst &basic_blocks) const
+{
+  if(i_it->is_assert())
```

Comment:
In the case where several instrumenters are being run, this will destroy the previous one's work. These may be redundant in the face of assert-to-assume always preceding instrumentation, or if this is needed for other reasons then we should simply rule out multiple coverage types for now.

Reply:
Well spotted.